pandas/core/reshape/merge.py

```diff
@@ -1708,6 +1708,22 @@ def _validate_specification(self):
         if self.left_by is not None and self.right_by is None:
             raise MergeError("missing right_by")
 
+        lo_dtype = (
```

Comment:
hmm i think this will just work in _get_merge_keys below no?

Reply:
add a comment before the clause on what you are checking

Reply:
Added a comment. Would not work out of the box in _get_merge_keys , because if self.left_index and self.right_index this is no merge key